mythril_multiboot2/src/services.rs

```diff
@@ -35,19 +35,30 @@ impl VmServices for Multiboot2Services {
     }
 
     fn rsdp(&self) -> Result<acpi::rsdp::RSDP> {
-        // Get a v1 tag if possible
-        self.info
-            .rsdp_v1_tag()
-            .map(|tag| {
-                // use empty string if no OEM id provided
-                let id = tag.oem_id().unwrap_or("      ").as_bytes();
+        self.info.rsdp_v2_tag().map_or_else(
+            || {
+                self.info.rsdp_v1_tag().map_or_else(
+                    || Err(Error::NotFound),
+                    |tag_v1| {
+                        let id = tag_v1.oem_id().unwrap_or("      ").as_bytes();
+                        let mut arr: [u8; 6] = [0; 6];
```

Comment:
You can move `arr` outside the closures. We only use this place in the leaves of the control flow graph, so something like the following should work.

```patch
diff --git a/mythril_multiboot2/src/services.rs b/mythril_multiboot2/src/services.rs
index 4d5ca7d..c209e9a 100644
--- a/mythril_multiboot2/src/services.rs
+++ b/mythril_multiboot2/src/services.rs
@@ -35,13 +35,13 @@ impl VmServices for Multiboot2Services {
     }
 
     fn rsdp(&self) -> Result<acpi::rsdp::RSDP> {
+        let mut arr: [u8; 6] = [0; 6];
         self.info.rsdp_v2_tag().map_or_else(
             || {
                 self.info.rsdp_v1_tag().map_or_else(
                     || Err(Error::NotFound),
-                    |tag_v1| {
+                    move |tag_v1| {
                         let id = tag_v1.oem_id().unwrap_or("      ").as_bytes();
-                        let mut arr: [u8; 6] = [0; 6];
                         arr.copy_from_slice(&id[0..6]);
                         Ok(acpi::rsdp::RSDP::V1 {
                             oemid: arr,
@@ -50,9 +50,8 @@ impl VmServices for Multiboot2Services {
                     },
                 )
             },
-            |tag_v2| {
+            move |tag_v2| {
                 let id = tag_v2.oem_id().unwrap_or("      ").as_bytes();
-                let mut arr: [u8; 6] = [0; 6];
                 arr.copy_from_slice(&id[0..6]);
                 Ok(acpi::rsdp::RSDP::V2 {
                     oemid: arr,
```

Reply:
🐒  i like to move it move it